### ISSUE TITLE
sandbox.resetHistory() doesn't reset history of spies

### ISSUE THREAD

**Issue by Flarna** [sinonjs/sinon#1372]
* Sinon version : 2.1.0
* Environment   : NodeJs 4.8.1
* Other libraries you are using: mocha

**What did you expect to happen?**
`sandbox.resetHistory()` should reset history of spies.

**What actually happens**
`sandbox.resetHistory()` is not reseting history of spies.

**How to reproduce**
```
it("reset spy", function(): void {
  const sb = Sinon.sandbox.create();
  const spy = sb.spy();
  spy();
  Assert.equal(spy.callCount, 1);
  spy();
  Assert.equal(spy.callCount, 2);
  sb.resetHistory();
  spy();
  Assert.equal(spy.callCount, 3);  // should fail but doesn't
  Assert.equal(spy.callCount, 1);  // should not fail but fails
});
```

**Comment by lucasfcosta** [comment]
Hi @Flarna, thanks for your issue!

You're using the wrong method to reset spies' history.

You should use `sandbox.reset()` instead of `sandbox.resetHistory().

According to the docs:

> `sandbox.resetHistory();`
> Resets the history of all **stubs** created through the sandbox

Since you want to restore **spies** and not stubs you should use:

> `sandbox.reset();`
> Resets the internal state of all **fakes** created through sandbox.

**Comment by Flarna** [comment]
If I use ``sandbox.reset()`` the history of spies is reseted but additionally the behavior of all stubs which is unfortunately not what I need.
I feel that resetting history and behavior should be separated everywhere.
Sure, spies have no behavior so I guess reset and resetHistory is effectively the same here.

**Comment by dslagle** [comment]
I just ran into this issue today and would agree with @Flarna. I expected sandbox.resetHistory() to also reset the history of the spies. It seems like overkill to need to use sandbox.reset() and then have to rebuild the behavior of all my stubs.

**Comment by mroderick** [comment]
The fix has been released as `v2.3.4`
